Description: Support Ubuntu langpacks
 Support alternative gettext tree in /usr/share/locale-langpack; if a file is
 present in both trees, prefer the newer one
Forwarded: not-needed
Origin: Debian cpython packaging
Author: Michael Vogt <michael.vogt@ubuntu.com>
Last-Update: 2011-12-19

--- a/lib-python/2.7/gettext.py
+++ b/lib-python/2.7/gettext.py
@@ -446,11 +446,26 @@
         if lang == 'C':
             break
         mofile = os.path.join(localedir, lang, 'LC_MESSAGES', '%s.mo' % domain)
+        mofile_lp = os.path.join("/usr/share/locale-langpack", lang,
+                               'LC_MESSAGES', '%s.mo' % domain)
+
+        # first look into the standard locale dir, then into the
+        # langpack locale dir
+
+        # standard mo file
         if os.path.exists(mofile):
             if all:
                 result.append(mofile)
             else:
                 return mofile
+
+        # langpack mofile -> use it
+        if os.path.exists(mofile_lp):
+            if all:
+                result.append(mofile_lp)
+            else:
+                return mofile_lp
+
     return result
 
 
